### PR TITLE
feat(welcome): Add media welcomes

### PR DESCRIPTION
**Pull request by EverythingSuckz**
- Add support to add custom media welcome messages

### CODE REVIEW

**Review by MrMissx — CHANGES_REQUESTED** [review]
Thank You
Please make sure everything are tested

File: anjani/language/id.yml [hunk 1]
```diff
@@ -569,8 +569,10 @@ bot-added: |
 default-welcome: Hai {first}, apa kabar?\nSelamat datang di {chatname}
 clean-serv-set: "Mengubah pembersihan pesan layanan : {}."
 welcome-set: Sambutan menjadi {} saat anggota baru bergabung.
+welcome-message-expired: File ID for the media message got expired.\n__This might have occured because you must have changed the bot token. Please re-add the welcome media.__
 cust-welcome-set: Berhasil mengubah pesan sambutan sesuai yang anda inginkan!
 reset-welcome: Berhasil mengubah pesan sambutan ke pesan sambutan bawaan saya!
+unsupported-media-command: "Media with commands as captions are not supported yet. **TIP**: __Maybe just reply the command to the media?__"
```
[the remark this answers is not in the document]
Comment (EverythingSuckz) [reply anjani/language/id.yml:575]: Oh okay, I thought it might break. I'll change it.

File: anjani/plugins/welcome.py [hunk 2]
```diff
@@ -109,26 +148,48 @@ async def _member_join(self, message: Message, reply_to: int) -> None:
...
-                    previous = await self.previous_welcome(chat.id, msg.id)
-                    if previous:
-                        try:
-                            await self.bot.client.delete_messages(chat.id, previous)
-                        except MessageDeleteForbidden:
-                            pass
+                    except MessageEmpty:
+                        self.log.warning("Welcome message empty on %s.", message.chat.id)
+
+                    if msg:
```
Comment (MrMissx) [inline anjani/plugins/welcome.py:186]: `msg` here are possibly unbound when it run the exception block

File: anjani/plugins/welcome.py [hunk 3]
```diff
@@ -183,13 +244,36 @@ async def is_goodbye(self, chat_id: int) -> bool:
...
         if message:
-            return message.get("custom_welcome"), message.get("button")
-
-        return await self.text(chat_id, "default-welcome", noformat=True), None
+            # This checks data for old welcome schema
+            # TODO: deprecate old schema on v3
+            if "custom_welcome" in message:
+                text: str = message["custom_welcome"]
+                button: Optional[Button] = message.get("button")
+                message_type: Types = Types.TEXT.value
```
Comment (MrMissx) [inline anjani/plugins/welcome.py:258]: No need to specify `x.value`  from an `Enum`. Just use `Types.TEXT`.

File: anjani/plugins/welcome.py [hunk 4]
```diff
@@ -183,13 +244,36 @@ async def is_goodbye(self, chat_id: int) -> bool:
 
     async def welc_message(
         self, chat_id: int
-    ) -> Tuple[Optional[str], Optional[Tuple[Tuple[str, str, bool]]]]:
+    ) -> Tuple[Optional[str], Optional[Tuple[Tuple[str, str, bool]]], Optional[int], Optional[str]]:
         """Get chat welcome string"""
-        message = await self.db.find_one({"chat_id": chat_id}, {"custom_welcome": 1, "button": 1})
+        message: Optional[Dict[str, Union[str, int, bool]]] = await self.db.find_one(
```
Comment (MrMissx) [inline anjani/plugins/welcome.py:249]: No need to strictly typed `find_one()`. The method already defined the return type `Optional[Mapping[str, Any]]`. It is good with just this
```python
message = await self.db.find_one(...)
```

**Review by MrMissx — CHANGES_REQUESTED** [review]
A few more several things that might cause a bug.
Other than that LGTM

File: anjani/plugins/welcome.py (hunk 2 above)
Comment (EverythingSuckz) [reply anjani/plugins/welcome.py:186]: What do you prefer? should I initialize the msg as None or should return a msg variable on exception.

Comment (MrMissx) [reply anjani/plugins/welcome.py:186]: better to declare it as `None`

**Review by MrMissx — APPROVED** [review]
ThankYou

**Pull request merged**